vscode-extension/package.json

```diff
@@ -35,6 +35,11 @@
           "default": true,
           "description": "Enable formatting with scalafmt"
         },
+        "metals.scalafmt.onSave": {
+          "type": "boolean",
+          "default": false,
+          "description": "Format file before saving it"
+        },
```

Comment:
this is a bit strange in VSCode (because there's already a `formatOnSave` option from VSCode itself), but I guess VSCode users can just ignore it.

Reply:
See https://github.com/scalameta/metals/pull/195#discussion_r167445526. There's a difference between them and we cannot rely on it, because it's VSCode specific.

A possible improvement to this situation: VSCode extension could check the state of these two options and warn user when both are on (which means that formatting (probably) will be done twice)